architecture/src/commonMain/kotlin/observable/FlowSubject.kt

```diff
@@ -65,8 +65,18 @@ open class StateFlowInitializedSubject<T>(
         coroutineScope,
         context,
         { observedStateFlow },
-        { observedStateFlow.value = it },
-        { observedStateFlow.value = it },
+        {
+            if (observation.currentOrNull != it) {
+                observation.setValue(ObservableOptional.Value(it))
+            }
+            observedStateFlow.value = it
+        },
+        {
+            if (observation.observedValue.valueOrNull != it) {
+                observation.observedValue = ObservableOptional.Value(it)
+            }
+            observedStateFlow.value = it
+        },
         observation,
     ) {
     init {
```

Comment:
What were the troubles overriding the `observedStateFlow` in the end?

Reply:
Difficult to do for the DefaultSubject since the stateFlow doesnt overlap. Also required a delay on iOS before posting to it since we need to wait for `observeFlow` to have started to prevent overwriting the data